### PR TITLE
Allow to select player class using mouse wheel

### PR DESCRIPTION
**Pull request by a1exsh**
Makes it easier to go back (and forward) through the player classes.

### CODE REVIEW

File: src/fheroes2/gui/player_info.cpp [hunk 1]
```diff
@@ -295,8 +356,21 @@ bool Interface::PlayersInfo::QueueEventProcessing( void )
...
+        }
+    }
+    else if ( le.MouseWheelDn() ) {
+        if ( nullptr != ( player = GetFromClassClick( le.GetMouseCursor() ) ) ) {
+            if ( conf.AllowChangeRace( player->GetColor() ) ) {
+                changeRaceToNext( *player );
+            }
+        }
+    }
+    else { // if ( le.MouseClickLeft() )
```
Comment (ihhub) [inline src/fheroes2/gui/player_info.cpp:373]: Please remove this commented code.

Comment (ihhub) [reply src/fheroes2/gui/player_info.cpp:373]: Actually it'll be even better if we add this condition explicitly.

File: src/fheroes2/gui/player_info.cpp [hunk 2]
```diff
@@ -295,8 +356,21 @@ bool Interface::PlayersInfo::QueueEventProcessing( void )
...
                 Font::BIG );
     }
-    // le.MouseClickLeft()
-    else {
+    else if ( le.MouseWheelUp() ) {
+        if ( nullptr != ( player = GetFromClassClick( le.GetMouseCursor() ) ) ) {
+            if ( conf.AllowChangeRace( player->GetColor() ) ) {
+                changeRaceToPrev( *player );
+            }
+        }
```
Comment (ihhub) [inline src/fheroes2/gui/player_info.cpp:364]: Please don't do two commands in one line (I totally understand it was copy-pasted from another code):
```cpp
        player = GetFromClassClick( le.GetMouseCursor() );
        if ( player != nullptr ) ) {
            if ( conf.AllowChangeRace( player->GetColor() ) ) {
                changeRaceToPrev( *player );
            }
        }
```
The same applies for the condition below.

File: src/fheroes2/gui/player_info.cpp [hunk 3]
```diff
@@ -32,6 +32,67 @@
...
+            player.SetRace( Race::WRLK );
+            break;
+        case Race::NECR:
+            player.SetRace( Race::WZRD );
+            break;
+        case Race::RAND:
+            player.SetRace( Race::NECR );
+            break;
+        default:
+            break;
```
Comment (ihhub) [inline src/fheroes2/gui/player_info.cpp:91]: I'd suggest to add an assertion here:
```cpp
// Did you add a new race? Add the logic above
assert( 0 );
```

File: src/fheroes2/gui/player_info.cpp [hunk 4]
```diff
@@ -32,6 +32,67 @@
...
+            player.SetRace( Race::NECR );
+            break;
+        case Race::NECR:
+            player.SetRace( Race::RAND );
+            break;
+        case Race::RAND:
+            player.SetRace( Race::KNGT );
+            break;
+        default:
+            break;
```
Comment (ihhub) [inline src/fheroes2/gui/player_info.cpp:62]: I'd suggest to add an assertion here:
```cpp
// Did you add a new race? Add the logic above
assert( 0 );
```

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @a1exsh , I put several comments in this pull request. Can you please check them once you have time?

File: src/fheroes2/gui/player_info.cpp (hunk 1 above)
Comment (a1exsh) [reply src/fheroes2/gui/player_info.cpp:373]: Was my idea as well, but we have a `const LocalEvent` here and somehow of all the checks that we do here `MouseClickLeft()` is not declared `const`.  I don't really want to open that can of worms :)

Comment (a1exsh) [reply src/fheroes2/gui/player_info.cpp:380]: @ihhub if you have ideas what to do here, please suggest

**Comment by ihhub** [comment]
@a1exsh , thank you so much for these changes!

**Pull request merged**